internal/converter/paths.go

```diff
@@ -14,44 +14,182 @@ import (
 					}
-					if item.Post != nil {
-						existing.Post = item.Post
-					}
-					if item.Delete != nil {
-						existing.Delete = item.Delete
-					}
-					if item.Put != nil {
-						existing.Put = item.Put
-					}
-					if item.Patch != nil {
-						existing.Patch = item.Patch
-					}
-					paths.PathItems.Set(path, existing)
+				}
+
+				// Update path items from google.api annotations
+				for pair := pathItems.First(); pair != nil; pair = pair.Next() {
+					updatePathItem(pair.Key(), pair.Value())
+				}
+
+				// Default to ConnectRPC/gRPC path if no google.api annotations
+				if pathItems == nil || pathItems.Len() == 0 {
+					path := "/" + string(service.FullName()) + "/" + string(method.Name())
+					updatePathItem(path, methodToPathItem(opts, method))
 				}
 			}
-			// No google.api annotations for this method, so default to the ConnectRPC/gRPC path
-			if pathItems == nil || pathItems.Len() == 0 {
-				paths.PathItems.Set("/"+string(service.FullName())+"/"+string(method.Name()), methodToPathItem(opts, method))
+		}
+
+	return nil
+}
+
+// Helper function to update operations
+func updateOperations(existing, newItem *v3.PathItem) {
+	operations := []struct {
+		existingOp **v3.Operation
+		newOp      *v3.Operation
+	}{
+		{&existing.Get, newItem.Get},
+		{&existing.Post, newItem.Post},
+		{&existing.Delete, newItem.Delete},
+		{&existing.Put, newItem.Put},
+		{&existing.Patch, newItem.Patch},
+	}
+
+	for _, op := range operations {
+		if op.newOp != nil {
+			*op.existingOp = op.newOp
+		}
+	}
+}
+
+func mergePathItems(existing, new *v3.PathItem) {
+	// Merge operations
+	if new.Get != nil {
+		mergeOperation(existing, new, &existing.Get, new.Get)
+	}
+	if new.Post != nil {
+		mergeOperation(existing, new, &existing.Post, new.Post)
+	}
+	if new.Put != nil {
+		mergeOperation(existing, new, &existing.Put, new.Put)
+	}
+	if new.Delete != nil {
+		mergeOperation(existing, new, &existing.Delete, new.Delete)
+	}
+	if new.Options != nil {
+		mergeOperation(existing, new, &existing.Options, new.Options)
+	}
+	if new.Head != nil {
+		mergeOperation(existing, new, &existing.Head, new.Head)
+	}
+	if new.Patch != nil {
+		mergeOperation(existing, new, &existing.Patch, new.Patch)
+	}
+	if new.Trace != nil {
+		mergeOperation(existing, new, &existing.Trace, new.Trace)
+	}
+
+	// Merge other fields
+	if new.Summary != "" {
+		existing.Summary = new.Summary
+	}
+	if new.Description != "" {
+		existing.Description = new.Description
+	}
+	existing.Servers = append(existing.Servers, new.Servers...)
+	existing.Parameters = append(existing.Parameters, new.Parameters...)
+
+	// Merge extensions
+	for pair := new.Extensions.First(); pair != nil; pair = pair.Next() {
+		if _, ok := existing.Extensions.Get(pair.Key()); !ok {
+			existing.Extensions.Set(pair.Key(), pair.Value())
+		}
+	}
+}
+
+func mergeOperation(existingItem, newItem *v3.PathItem, existing **v3.Operation, new *v3.Operation) {
+	if *existing == nil {
+		*existing = new
```

Comment:
Alternatively, you can return early instead of having a large 'else' block, like so:
```go
	if *existingPtr == nil {
		*existingPtr = new
		return
	}

	existing := *existingPtr
```
Also, it is a bit more readable if you assign `*existing` to a variable.